lms/djangoapps/courseware/management/commands/tests/test_dump_course.py

```diff
@@ -81,6 +86,14 @@ def call_command(self, name, *args, **kwargs):
         out.seek(0)
         return out.read()
 
+    def test_dump_course_ids_with_filter(self):
+        args = []
+        kwargs = {'end': '2013-12-01'}  # exclude any courses which have ended before 2013-12-01
+        output = self.call_command('dump_course_ids_with_filter', *args, **kwargs)
+        dumped_courses = output.strip().split('\n')
+        dumped_ids = set(dumped_courses)
+        assert {str(self.test_course_key)} == dumped_ids
```

Comment:
It looks like the test courses you set up would be excluded from the results of this management command, so I'm not sure why this test would pass. Am I reading it wrong?

Reply:
So there are two test courses that get created. In this instance only one of the two would get returned, the one with `end=TEST_COURSE_END,` on line #64.

Reply:
It's not trivial for me to understand where the other courses are getting created.  Can you add an explanation (in a docstring for this test) about the fact that there's logic somewhere else that adds more courses to the test database, and that this test is specifically not returning those other course(s)?